Lib/fontTools/misc/configTools.py

```diff
@@ -72,28 +74,29 @@ def __init__(self, name, value):
         )
 
 
-_NO_VALUE = object()
-
-
 class ConfigUnknownOptionError(ConfigError):
     """Raised when a configuration option is unknown."""
 
-    def __init__(self, name, value=_NO_VALUE):
-        super().__init__(
-            f"Config option {name} is unknown"
-            + ("" if value is _NO_VALUE else f" (with given value {repr(value)})")
+    def __init__(self, option_or_name):
+        name = (
+            f"'{option_or_name.name}' (id={id(option_or_name)})>"
+            if isinstance(option_or_name, Option)
+            else f"'{option_or_name}'"
         )
+        super().__init__(f"Config option {name} is unknown")
 
 
-@dataclass
+@dataclass(frozen=True)
 class Option:
+    name: str
+    """Unique name identifying the option (e.g. package.module:MY_OPTION)."""
     help: str
     """Help text for this option."""
     default: Any
     """Default value for this option."""
     parse: Callable[[str], Any]
     """Turn input (e.g. string) into proper type. Only when reading from file."""
-    validate: Callable[[Any], bool]
+    validate: Optional[Callable[[Any], bool]] = None
```

Comment:
You could also make the default value be lambda _: True?

Reply:
maybe, but I find a None more explicit for something that is mostly optional